src/dmp.rs

```diff
@@ -3041,6 +3042,64 @@ impl DiffMatchPatch {
         // Ok(html)
     }
 
+    /// Given a location in text1, compute and return the equivalent location in text2.
+    /// e.g. "The cat" -> "The big cat", 1->1, 4->8
```

Comment:
The doc comments in the original say `5->8` but that is inaccurate. It should actually be `4->8`. That's the result in the official implementation and that's what is correct.

```
❯ python
Python 3.9.6 (default, Nov 11 2024, 03:15:38) 
[Clang 16.0.0 (clang-1600.0.26.6)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
CPython >>> from diff_match_patch import diff_match_patch
CPython >>> dmp = diff_match_patch()
CPython >>> diffs = dmp.diff_main("the cat", "the big cat")
CPython >>> dmp.diff_xIndex(diffs, 4)
8
CPython >>> dmp.diff_xIndex(diffs, 5)
9
```